core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/WorkflowCompiler.scala

```diff
@@ -20,12 +20,14 @@ import scala.collection.{JavaConverters, mutable}
 
 class WorkflowCompiler(val workflowInfo: WorkflowInfo, val context: WorkflowContext) {
 
+  init()
```

Comment:
just want to make sure this works like calling the `init()` function inside constructor in java, the syntax is a bit confusing in scala and I'm not used to it.

Reply:
yes, it is confusing that scala does not give default constructor interface, but yes here I believe it is invoked in constructor. @shengquan-ni can confirm?